pilot/pkg/config/kube/ingress/conversion.go

```diff
@@ -263,6 +276,28 @@ func ingressBackendToHTTPRoute(backend *v1beta1.IngressBackend, namespace string
 	}
 }
 
+func resolveNamedPort(backend *v1beta1.IngressBackend, namespace string, serviceLister listerv1.ServiceLister,
+	podLister listerv1.PodLister) (int32, error) {
+	svc, err := serviceLister.Services(namespace).Get(backend.ServiceName)
+	if err != nil {
+		return 0, err
+	}
+	podList, err := podLister.Pods(namespace).List(k8sLabels.Set(svc.Spec.Selector).AsSelector())
+	if err != nil {
+		return 0, err
+	}
+	if len(podList) > 0 {
+		for _, container := range podList[0].Spec.Containers {
+			for _, port := range container.Ports {
+				if port.Name == backend.ServicePort.StrVal {
```

Comment:
Sorry I didn't catch this earlier.. I am pretty sure we are referencing the wrong object. I think we are supposed to match a named port in the *service*, not the pod. From the spec ` Specifies the port of the referenced service.` and the name is ServicePort.

This makes sense as well, since this wouldn't result in correct behavior (two pods may match the service but have different port definitions).

So I think we can get rid of all the pod related stuff, and just search within `svc.Ports` here

Reply:
My bad, somehow got confused between them 😅 
I made the changes and also added an integration test which tests ingress with named port.